include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,184 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+            : m_domain(impl.m_domain)
+            , m_nknots(impl.m_nknots)
+        {
+        }
+
+        /// @brief Copy-constructs
         Impl(Impl const& x) = default;
 
+        /// @brief Move-constructs
         Impl(Impl&& x) = default;
 
+        /// @brief Destructs
         ~Impl() = default;
 
+        /// @brief Copy-assigns
         Impl& operator=(Impl const& x) = default;
 
+        /// @brief Move-assigns
         Impl& operator=(Impl&& x) = default;
 
+        /** @brief Evaluates B-splines at a given coordinate
+         *
+         * The values are computed for every B-splines at the given coordinate x. A spline approximation at coordinate x is a linear combination of those B-splines evaluations weigthed with splines coefficients of the spline-transformed initial discrete function.
+         * @param[out] values The values of the B-splines evaluated at coordinate x. It has to be a (1+degree) 1D mdspan.
+         * @param[in] x The coordinate where B-splines are evaluated.
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_basis(DSpan1D values, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates B-splines derivatives at a given coordinate
+         *
+         * The derivatives are computed for every B-splines at the given coordinate x. A spline approximation of a derivative at coordinate x is a linear combination of those B-splines derivatives weigthed with splines coefficients of the spline-transformed initial discrete function.
+         * @param[out] derivs The derivatives of the B-splines evaluated at coordinate x. It has to be a (1+degree) 1D mdspan.
+         * @param[in] x The coordinate where B-splines derivatives are evaluated.
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_deriv(DSpan1D derivs, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates B-splines values and n derivatives at a given coordinate
+         *
+         * The values and derivatives are computed for every B-splines at the given coordinate x.
+         * @param[out] derivs The values and n derivatives of the B-splines evaluated at coordinate x. It has to be a (1+degree)*(1+n) 2D mdspan.
+         * @param[in] x The coordinates where B-splines derivatives are evaluated.
+         * @param[in] n The number of derivatives to evaluate (in addition to the B-splines values themselves).
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type eval_basis_and_n_derivs(
                 ddc::DSpan2D derivs,
                 ddc::Coordinate<Tag> const& x,
                 std::size_t n) const;
 
+        /** @brief Compute the integrals of the B-splines.
+         *
+         * @param[out] int_vals The values of the integrals. It has to be a (1+nbasis) 1D mdspan.
+         * @return The values of the integrals.
+         */
```

Comment:
```suggestion
        /** @brief Compute the integrals of the B-splines.
         *
         * The integral of each of the B-splines over their support within the domain on which this basis was defined.
         *
         * @param[out] int_vals The values of the integrals. It has to be a (1+nbasis) 1D mdspan.
         * @return The values of the integrals.
         */
```
> over their support within the domain on which this basis was defined.

This is especially important to mention for the uniform case where B-splines have support outside the domain